app/src/routes/v3/layer.router.js

```diff
@@ -0,0 +1,162 @@
+const Router = require("koa-router");
+const logger = require("logger");
+const LayerModel = require("models/layer.model");
+const LayerSerializer = require("serializers/layer.serializer");
+const UserMiddleware = require("middleware/user.middleware");
+const LayerService = require("services/layer.service");
+const LayerValidator = require("validators/layer.validator");
+const TeamService = require("services/team.service");
+//const lossLayerProvider = require("lossLayer.provider");
+//const TileNotFoundError = require("TileNotFoundError");
+const V3TeamService = require("../../services/v3TeamService");
+const V3LayerService = require("../../services/v3LayerService");
+
+const router = new Router({
+  prefix: "/contextual-layer"
+});
+
+class Layer {
+  static get middleware() {
+    return [UserMiddleware.mapAuthToUser];
+  }
+
+  static async createTeamLayer(ctx) {
+    logger.info("Create team layer");
+    const owner = { type: LayerService.type.TEAM, id: ctx.params.teamId };
+    let team = null;
+    try {
+      team = await TeamService.getTeam(owner.id);
+    } catch (e) {
+      logger.error(e);
+      ctx.throw(500, "Team retrieval failed.");
+    }
+    // get list of user teams
+    let teams = await V3TeamService.getUserTeams(ctx.request.body.user.id);
+    let matchingTeam = teams.find(obj => obj.id === owner.id);
+    const isManager =
+      matchingTeam &&
+      matchingTeam.attributes &&
+      (matchingTeam.attributes.userRole.toString() === "manager" ||
```

Comment:
could write `matchingTeam?.attributes?.userRole?.toString() === "manager" || matchingTeam?.attributes?.userRole?.toString() === "administrator"`

[https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)

Reply:
I've never been able to get this to work. Always throws an error

Details:

    /opt/fw-contextual-layer/app/src/routes/v3/layer.router.js:37
          (matchingTeam?.attributes?.userRole?.toString() === "manager" ||
                        ^

    SyntaxError: Unexpected token .

Reply:
That's annoying, no worries then!